lib/app.js

```diff
@@ -0,0 +1,69 @@
+const HOOK_STAGE_PRE_INSTALL = 'preInstall'
+const HOOK_STAGE_POST_INSTALL = 'postInstall'
+
+const isObject = o => typeof o === 'object' && o !== null && !Array.isArray(o)
+
+const runHandler = handler => {
+  handler = typeof handler === 'function' ? handler : () => {}
+
+  let resultMaybePromise
+  try {
+    resultMaybePromise = handler()
+  } catch (err) {
+    resultMaybePromise = false
```

Comment:
Could short circuit again here:
```suggestion
    return Promise.resolve(false);
```

Reply:
done